javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt

```diff
@@ -107,6 +112,93 @@ class StorageServiceTest {
         assertThat(receivedIds[0]).isEqualTo("test")
     }
 
+    @Test
+    fun testPeriodicJobsConfig_Default() = runBlocking {
+        StorageService().onCreate()
+
+        workManager.assertEnqueued("PeriodicCleanupTask")
+        workManager.assertEnqueued("DatabaseGarbageCollectionPeriodicTask")
+    }
+
+    @Test
+    fun testPeriodicJobsConfig_Subclass_Disabled() = runBlocking {
+        class MyStorageService : StorageService() {
+            override val config = StorageServiceConfig(false, 1, false, 1)
+        }
+        MyStorageService().onCreate()
+
+
+        workManager.assertNotEnqueued("PeriodicCleanupTask")
+        workManager.assertNotEnqueued("DatabaseGarbageCollectionPeriodicTask")
+    }
+
+    @Test
+    fun testPeriodicJobsConfig_Subclass_OneOneOneOff() = runBlocking {
+        class MyStorageService : StorageService() {
+            override val config = StorageServiceConfig(false, 1, true, 1)
+        }
+        MyStorageService().onCreate()
+
+        workManager.assertNotEnqueued("PeriodicCleanupTask")
+        workManager.assertEnqueued("DatabaseGarbageCollectionPeriodicTask")
+    }
+
+    @Test
+    fun testPeriodicJobsConfig_Subclass_StopStart() = runBlocking {
+        class MyStorageService : StorageService() {
+            fun changeConfig(config: StorageServiceConfig) = schedulePeriodicJobs(config)
+        }
+        val sts = MyStorageService()
+        sts.onCreate()
+
+        workManager.assertEnqueued("PeriodicCleanupTask")
+        workManager.assertEnqueued("DatabaseGarbageCollectionPeriodicTask")
+
+        sts.changeConfig(StorageService.StorageServiceConfig(false, 2, false, 2))
+
+        workManager.assertCanceled("PeriodicCleanupTask")
+        workManager.assertCanceled("DatabaseGarbageCollectionPeriodicTask")
+
+        sts.changeConfig(StorageService.StorageServiceConfig(true, 2, false, 2))
+
+        workManager.assertEnqueued("PeriodicCleanupTask")
+        workManager.assertCanceled("DatabaseGarbageCollectionPeriodicTask")
+    }
+
+    @Test
+    fun testPeriodicJobsConfig_Subclass_DisableAll() = runBlocking {
+        class MyStorageService : StorageService() {
+            fun disableAll() = disableAllPeriodicJobs()
+        }
+        val sts = MyStorageService()
+        sts.onCreate()
+
+        workManager.assertEnqueued("PeriodicCleanupTask")
+        workManager.assertEnqueued("DatabaseGarbageCollectionPeriodicTask")
+
+        sts.disableAll()
+
+        workManager.assertCanceled("PeriodicCleanupTask")
+        workManager.assertCanceled("DatabaseGarbageCollectionPeriodicTask")
+    }
+
+    private fun WorkManager.assertEnqueued(tag: String) {
```

Comment:
Optional: you could simplify this a little by just making it a regular method and operate on the workManager member variable. I don't think you ever define a new WorkManager variable.